### PR TITLE
Relocate backlight drivers

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

- Moved PWM drivers to `platforms/<platform>/drivers` and renamed to `backlight_pwm`
- Moved timer driver to ChibiOS platform dir
- Removed quantum.h includes
- Fixed incorrect PAL mode for ChibiOS PWM driver (should be `2` for `B8` on Proton-C; also per docs)

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fauxpark** [comment]
CI failure is unrelated.

As for SHA mismatches, there are a ton. Not sure why. I'm expecting some to possibly come from the `BACKLIGHT_PAL_MODE` default change, but the bulk appear to be AVR boards.

File: builddefs/common_features.mk [hunk 1]
```diff
@@ -573,11 +574,7 @@ ifeq ($(strip $(BACKLIGHT_ENABLE)), yes)
         OPT_DEFS += -DBACKLIGHT_CUSTOM_DRIVER
     else
         SRC += $(QUANTUM_DIR)/backlight/backlight_driver_common.c
-        ifeq ($(strip $(BACKLIGHT_DRIVER)), pwm)
-            SRC += $(QUANTUM_DIR)/backlight/backlight_$(PLATFORM_KEY).c
-        else
-            SRC += $(QUANTUM_DIR)/backlight/backlight_$(strip $(BACKLIGHT_DRIVER)).c
-        endif
+        SRC += backlight_$(strip $(BACKLIGHT_DRIVER)).c
```
Comment (drashna) [inline builddefs/common_features.mk:577]: Not a fan of having a relative path.  They tend to generate build objects in less than ideal places.

But not sure there is a better way to handle this, other than with a bunch of if's

Comment (fauxpark) [reply builddefs/common_features.mk:577]: Yeah, I wasn't completely happy with that, but I think I can make it slightly better.

**Review by drashna — APPROVED** [review]

File: builddefs/common_features.mk (hunk 1 above)
Comment (drashna) [reply builddefs/common_features.mk]: Change isn't great, but at least build artifacts should be handled better. 👍🏻

**Review by tzarc — APPROVED** [review]

**Pull request merged**